docs/tutorials/13_quantum_random_access_optimizer.ipynb

```diff
@@ -0,0 +1,736 @@
+     "output_type": "stream",
+     "text": [
+      "objective function value: 9.0\n",
+      "variable values: x_0=0.0, x_1=1.0, x_2=0.0, x_3=1.0, x_4=1.0, x_5=0.0\n",
+      "status: SUCCESS\n"
+     ]
+    }
+   ],
+   "source": [
+    "from qiskit.algorithms.minimum_eigensolvers import NumPyMinimumEigensolver\n",
+    "\n",
+    "from qiskit_optimization.algorithms import MinimumEigenOptimizer\n",
+    "\n",
+    "exact_mes = NumPyMinimumEigensolver()\n",
+    "exact = MinimumEigenOptimizer(exact_mes)\n",
+    "exact_result = exact.solve(problem)\n",
+    "print(exact_result.prettyprint())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The approximation ratio (QRAO's approximate optimal function value divided by the exact optimal function value) tells us how closely QRAO approximated the optimal solution to the problem."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "QRAO Approximate Optimal Function Value: 5.0\n",
+      "Exact Optimal Function Value: 9.0\n",
+      "Approximation Ratio: 0.56\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"QRAO Approximate Optimal Function Value:\", results.fval)\n",
+    "print(\"Exact Optimal Function Value:\", exact_result.fval)\n",
+    "print(f\"Approximation Ratio: {results.fval /  exact_result.fval :.2f}\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Solve the problem using the `QuantumRandomAccessOptimizer` with `MagicRounding`\n",
+    "\n",
+    "Magic rounding is a quantum technique employed to map the ground state results of our encoded Hamiltonian back to a solution of the original problem. Unlike semideterministic rounding, magic rounding requires a quantum backend, which can be either hardware or a simulator. The backend is passed to the MagicRounding class through a `Sampler`, which also determines the total number of shots (samples) that magic rounding will utilize. Consequently, we need to specify `Estimator` and `Sampler` for the optimizer and the rounding scheme, respectively.\n",
+    "\n",
+    "In practice, users may choose to set a significantly higher number of magic rounding shots compared to the shots used by the minimum eigensolver for the relaxed problem. This difference arises because the minimum eigensolver estimates expectation values, while the magic rounding scheme returns the sample corresponding to the maximum function value found. When estimating an expectation value, increasing the number of shots enhances the convergence to the true value. However, when aiming to identify the largest possible function value, we often sample from the tail of a distribution of outcomes. As a result, until we observe the highest value outcome in our distribution, each additional shot increases the expected return value.\n",
```

Comment:
hmm, I've never heard of any rule of thumb.

Reply:
You understood what I meant I guess by 'rule of thumb' - I didn't think when I used that phrase. I just meant some general guideline for users so that `users may choose to set a significantly higher number of magic round shots` might be a little more concrete for someone reading it. As I noted I saw there a factor of 10 was chosen - maybe that's a reasonable `significantly higher number`?

Reply:
I see. hmm, `users may choose to set a significantly higher number of magic round shots` + `a factor of 10` might confuse users.  I will add more texts.

When we're dealing with `basis_sampling` set to `uniform`, the number of `magic round shots` directly impacts the diversity of the computational basis you can generate. In other words, it's about the variety of distinct bases we can get from a quantum computer.

For example, if you have a quantum computer with 20 qubits, it can potentially generate 1 million different computational bases. Now, if you set the `magic round shots` to 10,000, then from this pool of 1 million, you can only obtain at most 10,000 distinct bases. When the optimal solution is in the other 990,000 basis, we can’t obtain it.

That's why it's vital to understand the relationship between the number of `magic round shots` and the diversity of the computational basis your quantum computer can generate.